Tests/AppTests/IngestorTests.swift

```diff
@@ -83,12 +83,17 @@ class IngestorTests: AppTestCase {
         // MUT
         await ingest(client: app.client, database: app.db, packages: packages)
 
-        // validate the second package's license is updated
-        let repo = try await Repository.query(on: app.db)
-            .filter(\.$name == "2")
-            .first()
-            .unwrap()
-        XCTAssertEqual(repo.licenseUrl, "license")
+        do {
+            // validate the second package's license is updated
+            let repo = try await Repository.query(on: app.db)
+                .filter(\.$name == "2")
+                .first()
+                .unwrap()
+            XCTAssertEqual(repo.licenseUrl, "license")
+            for pkg in try await Package.query(on: app.db).all() {
+                XCTAssertEqual(pkg.processingStage, .ingestion, "\(pkg.url) must be in ingestion")
```

Comment:
This is a crucial bit of additional test coverage where we ensure the processing stage is advanced when there are errors.